File: scripts/workspace/orchestrator/probes/vault-probe.py
#!/usr/bin/env python3
"""vault-probe.py — check that each KEY argument resolves to a non-empty
value in os.environ. Emits one JSON line per key on stdout.

DEVIATION FROM PLAN: the plan called this a "vault probe" against a vault
module. The codebase has no vault module — env keys come straight from
os.environ. This probe checks that source directly.

Usage: python vault-probe.py KEY1 KEY2 KEY3 ...
"""
import json
import os
import sys

for key in sys.argv[1:]:
    v = os.environ.get(key)
    ok = isinstance(v, str) and len(v) > 0
    print(json.dumps({
        "key": key,
        "twin": "fastapi",
        "ok": ok,
        "value_length": len(v) if ok else None,
    }))

File: scripts/workspace/orchestrator/probes/vault-probe.mjs
#!/usr/bin/env node
// vault-probe.mjs — check that each KEY argument resolves to a non-empty
// value in process.env. Emits one JSON line per key on stdout.
//
// DEVIATION FROM PLAN: the plan called this a "vault probe" against a vault
// module. The codebase has no vault module — env keys come straight from
// process.env. This probe checks that source directly.
//
// Usage: node vault-probe.mjs KEY1 KEY2 KEY3 ...

for (const key of process.argv.slice(2)) {
  const v = process.env[key];
  const ok = typeof v === "string" && v.length > 0;
  process.stdout.write(
    JSON.stringify({
      key,
      twin: "fastify",
      ok,
      value_length: ok ? v.length : null,
    }) + "\n",
  );
}
